sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -971,9 +971,36 @@ object TypeCoercion {
         case (ArrayType(fromType, true), ArrayType(toType: DataType, false)) => null
 
         case (ArrayType(fromType, false), ArrayType(toType: DataType, false))
-            if !Cast.forceNullable(fromType, toType) =>
+          if !Cast.forceNullable(fromType, toType) =>
           implicitCast(fromType, toType).map(ArrayType(_, false)).orNull
 
+        // Implicit cast between Map types.
```

Comment:
Do we have enough test for this new implicit cast rules?

Reply:
@viirya actually i followed the arraytype example. Actually it has decent coverage, i think. If you can think of a case thats missing , i will add. I will also go through once again myself.